cachito/jenkins-scripts/fetch-failed-request-logs.py

```diff
@@ -0,0 +1,195 @@
+    cachito_total_requests = []
+    api_url = "http://" + cachitoserver + "/" + API_PATH
+
+    # Initial URL.  The meta section of the response gives the 'next' URL with
+    # an updated page value, and 'None' when at the end of the available
+    # requests based on the query.  Using the 'next' URL to simplify the logic here.
+    request_url = (
+        api_url
+        + "/requests"
+        + "?per_page="
+        + str(per_page_requests)
+        + "&page="
+        + str(page)
+        + "&created_from="
+        + str(request_date)
+        + "&created_to="
+        + str(request_date)
+    )
+    while True:
+        r = http.request("GET", ic(request_url))
+        json_response = json.loads(r.data)
+        ic(json_response)
+
+        cachito_total_requests.extend(
+            [item for item in json_response["items"] if _is_same_date(item, request_date)]
+        )
+        cachito_failed_requests.extend(
+            [
+                item
+                for item in json_response["items"]
+                if _is_same_date(item, request_date) and item["state"] == state
+            ]
+        )
+        if json_response["meta"]["next"] is None:
+            ic("Received None for next URL... breaking")
+            break
+        request_url = (
+            json_response["meta"]["next"]
+            + "&created_from="
+            + str(request_date)
+            + "&created_to="
+            + str(request_date)
+        )
+
+    ic(cachito_total_requests)
+    ic(cachito_failed_requests)
+
+    return (
+        cachito_failed_requests,
+        len(cachito_failed_requests),
+        len(cachito_total_requests),
+    )
+
+
+def _fetch_cachito_logs(cachitoserver, id):
+    #
+    # Fetch logs for a specified request.
+    #
+    ic("Fetching logs for id " + str(id))
+    http = urllib3.PoolManager()
+    api_url = "https://" + cachitoserver + "/" + API_PATH
+    r = http.request("GET", api_url + "/requests/" + str(id) + "/logs")
+    ic(str(r.data.decode("utf-8")))
+
+    if not os.path.isdir("logs"):
+        raise Exception("logs directory is not a directory or does not exist")
+
+    f = open("logs/cachito-log-" + str(id) + ".txt", "w")
+    f.write(str(r.data.decode("utf-8")))
+    f.close()
+
+
+def main() -> int:
+    """
+    cachito-fail.py - Fetch all failed Cachito requests for the day specfied.
+
+    :return:
+        Exit 0 on success
+    """
+    args = _parse_cli()
+    ic(args)
+
+    if args.yesterday is False and args.date is None:
+        sys.exit("Either a --date or --yesterday argument is required...")
+
+    if args.yesterday:
+        request_date = date.today() - timedelta(days=1)
+    else:
+        request_date = datetime.datetime.strptime(args.date, "%Y-%m-%d").date()
+
+    if (date.today() - timedelta(days=7)) > request_date:
+        print("Logs are not available for jobs more than 7 days old")
+        sys.exit()
+
+    ic(request_date)
+
+    cachitoserver = args.cachitoserver
```

Comment:
This argument should be validated to avoid `None` values, since these will cause a concatenation error.